### ISSUE TITLE
Improve error handling of `TaggedTable`

### ISSUE THREAD

**Issue by lars-reimann** [Safe-DS/Stdlib#150]
### Is your feature request related to a problem?

In some cases, error handling is not ideal (see solution for details).

### Desired solution

* Validate inputs of a function as the first step
* Raise an appropriate exception with an appropriate message if issues are found
* Document in the docstring that the exception is raised and under which condition
* Test that the correct exception with the correct message is raised

**Comment by alex-senger** [comment]
Blocked by #58

**Comment by guenterk** [comment]
I do not see why #58 blocks this issue. Only testing should be blocked until #58 is completed, to avoid useless double work. There is no reason why #58 prevents input validation, raising exceptions and documenting them. If all of that that is already implemented, mention it in the comment.

**Comment by zzril** [comment]
Currently, `TaggedTable` has barely any methods yet.

There are the constructor-like methods `_from_table()` and `__init__()`, which do some sort of input validation. (Making sure that the `target` column won't be a `feature` column at the same time; not sure what else to validate here?)  
The only other methods are mere getters (`features` and `target`).

So, currently there's not much (possibly nothing?) to do in terms of "improving error handling" / input validation.

With #58 done, `TaggedTable` will have a whole lot of new methods, namely those ones inherited (and overridden) from `Table`.  
The solution implemented in the current PR for #58 (#332) does some input validation for these new methods already.

Once that PR is merged, once can think about refining the error handling. (Or alternatively make it part of #58 and drop this separate issue altogether.)
